### PR TITLE
Add cron start/stop events to beholder queue, resolves SYN-4311

### PR DESCRIPTION
**Pull request by rakuy0**

### CODE REVIEW

File: synapse/tests/test_lib_agenda.py [hunk 1]
```diff
@@ -723,7 +724,7 @@ async def test_agenda_custom_view(self):
             nodes = await core.nodes('test:int=97', opts={'view': newview})
             self.len(0, nodes)
 
-    async def test_agenda_custom_view(self):
+    async def test_agenda_edit_creator(self):
```
Comment (rakuy0) [inline synapse/tests/test_lib_agenda.py:727]: A found copy/paste error

File: synapse/lib/agenda.py [hunk 2]
```diff
@@ -886,6 +891,7 @@ async def _runJob(self, user, appt):
                 success = True
                 result = f'finished successfully with {count} nodes'
             finally:
+                await self.core.feedBeholder('cron:stop', {'iden': appt.iden})
```
Comment (vEpiphyte) [inline synapse/lib/agenda.py:894]: is this safe to do if we've been fini'd?

Comment (rakuy0) [reply synapse/lib/agenda.py:894]: Probably not. I'll add some logic.

File: synapse/lib/agenda.py [hunk 3]
```diff
@@ -886,6 +891,8 @@ async def _runJob(self, user, appt):
                 success = True
                 result = f'finished successfully with {count} nodes'
             finally:
+                if not self.isfini:
+                    await self.core.feedBeholder('cron:stop', {'iden': appt.iden})
```
Comment (vEpiphyte) [inline synapse/lib/agenda.py:895]: couldn't this just go into the other isfini check?

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**